pkg/postgres/repository.go

```diff
@@ -79,30 +83,55 @@ func ParseRunGroupIdentifier(id string) (dir martaapi.Direction, line martaapi.L
 	return martaapi.Direction(parts[0]), martaapi.Line(parts[1]), parts[2]
 }
 
-//EnsureTables ensures that all necessary tables exist
-func (a *RepositoryAgent) EnsureTables() error {
-	_, err := a.DB.Exec(`
+//EnsureTables ensures that all necessary tables exist. Three fields (line_id,
+//station_id, and direction_id) are always included, however if thirdRail is
+//false, they are always left empty. Including them in both cases simplifies
+//our update/select queries.
+func (a *RepositoryAgent) EnsureTables(thirdRail bool) error {
+	runsExtras := `
+	line_id integer,
+	direction_id integer,
+`
+	if thirdRail {
+		runsExtras = `
+	line_id integer REFERENCES lines(id),
+	direction_id integer REFERENCES directions(id),
```

Comment:
Always include the field, so that we don't need to chop fields out of our SELECT statements, but only include the foreign key reference if we're in third-rail mode.